app/src/main/java/de/westnordost/streetcomplete/quests/building_type/AddBuildingTypeForm.java

```diff
@@ -0,0 +1,170 @@
+			GARAGES =		new Item("garages", R.drawable.building_garages, R.string.quest_buildingType_garages),
+			PARKING = 		new Item("parking", R.drawable.building_parking, R.string.quest_buildingType_parking),
+
+			BARN =			new Item("barn", R.drawable.building_barn,	R.string.quest_buildingType_barn),
+			COWSHED =		new Item("cowshed", R.drawable.building_cowshed,	R.string.quest_buildingType_cowshed),
+			STABLE =		new Item("stable", R.drawable.building_stable,	R.string.quest_buildingType_stable),
+
+			SHED =			new Item("shed", R.drawable.building_shed,	R.string.quest_buildingType_shed),
+			HUT =			new Item("hut", R.drawable.building_hut, R.string.quest_buildingType_hut),
+			ROOF =			new Item("roof", R.drawable.building_roof, R.string.quest_buildingType_roof),
+			GREENHOUSE =	new Item("greenhouse",	R.drawable.building_greenhouse,	R.string.quest_buildingType_greenhouse),
+			CABIN =			new Item("cabin", R.drawable.building_cabin, R.string.quest_buildingType_cabin),
+			CONSTRUCTION =	new Item("construction", R.drawable.building_construction, R.string.quest_buildingType_construction),
+			RUINS =			new Item("ruins", R.drawable.building_ruins, R.string.quest_buildingType_ruins),
+			SERVICE =		new Item("service", R.drawable.building_service, R.string.quest_buildingType_service),
+			TRANSFORMER_TOWER = new Item("transformer_tower", R.drawable.building_transformer_tower, R.string.quest_buildingType_transformer_tower);
+
+	private final Item[] TOP_BUILDINGS = new Item[] {
+			HOUSE, DETACHED, APARTMENTS,
+			GARAGE, GARAGES, HUT,
+			RETAIL, INDUSTRIAL
+	};
+
+	private final Item[] ALL_BUILDINGS = new Item[] {
+			new Item("residential", R.drawable.panorama_building_residential, R.string.quest_buildingType_residential, new Item[]{
+					DETACHED, HOUSE, TERRACE, APARTMENTS, BUNGALOW,
+					FARM, HOUSEBOAT, STATIC_CARAVAN, DORMITORY, HOTEL,
+			}),
+			new Item("commercial", R.drawable.panorama_building_commercial, R.string.quest_buildingType_commercial, new Item[]{
+					INDUSTRIAL, RETAIL, KIOSK, WAREHOUSE
+			}),
+			new Item("civic", R.drawable.panorama_building_civic, R.string.quest_buildingType_civic, new Item[]{
+					COLLEGE, UNIVERSITY, SCHOOL, PUBLIC, HOSPITAL, STADIUM, TRAIN_STATION, TRANSPORTATION
+			}),
+			new Item(null, R.drawable.panorama_building_religious, R.string.quest_buildingType_religious, new Item[]{
+					CHURCH, CATHEDRAL, CHAPEL, MOSQUE, TEMPLE, SYNAGOGUE, SHRINE
+			}),
+			new Item(null, R.drawable.panorama_building_cars, R.string.quest_buildingType_cars, new Item[]{
+					CARPORT, GARAGE, GARAGES, PARKING
+			}),
+			new Item("farm_auxiliary", R.drawable.panorama_building_farm_auxiliary, R.string.quest_buildingType_farm_auxiliary, new Item[]{
+					BARN, COWSHED, STABLE
+			}),
+			new Item(null, R.drawable.panorama_building_other, R.string.quest_buildingType_other, new Item[]{
+					HUT, SHED, CABIN, GREENHOUSE, CONSTRUCTION, RUINS, ROOF, SERVICE, TRANSFORMER_TOWER
```

Comment:
- Transformer tower seems to be awfully specific. I think this should go
- What is a "service building"? It must be better defined what makes a service building
- I think ruins should go. No building has been constructed as a ruin, it seems to me that it should be a property of a building, not a "building type". Also, till when is a building still not-a-ruin and starting when is a building a ruin?
- greenhouse should go into the farm category
- hut, cabin, shed: As long as it is not clear what's the definite difference between them, they shouldn't be shown

Reply:
Hut/cabin - I would expect them to be intended to be inhabited (show with windows?), I would expect shed to be built to be used solely for storage

Reply:
> Transformer tower seems to be awfully specific.

Will be removed with the next commit

> What is a "service building"? It must be better defined what makes a service building

I think this will be done with the redesign of this quest...

> greenhouse should go into the farm category

Done with the next commit